clients/include/rocblas_arguments.hpp

```diff
@@ -324,4 +328,216 @@ static_assert(std::is_trivial<Arguments>{},
               "Arguments is not a trivial type, and thus is "
               "incompatible with C.");
 
+enum arg_type
```

Comment:
An `enum` with another set of names to maintain, makes maintenance harder.

Better to use `std::tuple` to list name-value pairs, and pass the tuples to a function which prints the elements out. Then if a particular benchmark needs a particular set of `Arguments`, it just lists them by name, value pairs, adjusting the values as necessary, and passes them to a universal print routine. There's no need to maintain a `vector` of arguments, a list of `enums` of arguments, a list of macros for generate `CASE`, etc. Just pass name-value pairs to a function with formats the output you way you want. That's what we do in logging today.

Reply:
Or if you really want 100% dynamic name lookup, without any duplication of "name", name, then create a separate `Arguments` data structure which uses `std:map` or `std::unordered_map` to map names to fields in `Arguments`. The creation of the map, as well as other ordered operations on "name", value pairs, can be automated in an `Arguments` and can work to construct a `std::map` or to print out elements -- it simply creates a `std::tuple` of every "name", value pair in `Arguments`, which can be then be printed out or transformed into a `std::map`.

Reply:
This class is to eliminate the typing of strings and argument ordering, using just the enums ensure no typos.   The logging is also a form of duplication which could be potentially eliminated and handled by a helper class(es).

Reply:
The new logging code will have everything listed once in one place -- there is no need for this solution.

This is attempting a C++ solution for something better handled in Python.